### ISSUE TITLE
self-hosted option in kubeadm fails

### ISSUE THREAD

**Issue by andrewrynhard** [kubernetes/kubernetes#43949]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.0", GitCommit:"fff5156092b56e6bd60fff75aad4dc9de6b6ef37", GitTreeState:"clean", BuildDate:"2017-03-28T16:36:33Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

**What happened**:
Bootstrapping a cluster with the patched version of `kubeadm` (https://github.com/kubernetes/kubernetes/pull/43835), the self hosted option fails with:
```
[self-hosted] self-hosted kube-apiserver ready after 16.057699 seconds
failed to create self-hosted "kube-scheduler" deployment [deployments.extensions "self-hosted-kube-scheduler" is forbidden: not yet ready to handle request]
```

**What you expected to happen**:
self-hosted scheduler should start.

**How to reproduce it** (as minimally and precisely as possible):
Follow the usual installation options for kubeadm, but update kubeadm to a patched version:
```
curl -o /bin/kubeadm https://storage.googleapis.com/kubernetes-release-dev/ci/v1.6.1-beta.0.12+018a96913f57f9/bin/linux/amd64/kubeadm
```

```
cat <<EOF > kubeadm.yaml 
apiVersion: kubeadm.k8s.io/v1alpha1
kind: MasterConfiguration
kubernetesVersion: v1.6.0
selfHosted: true
authorizationMode: RBAC
token: e3f816.c0212916116e6e48
networking:
  dnsDomain: cluster.local
  serviceSubnet: 10.96.0.0/12
  podSubnet: 10.244.0.0/16
EOF
kubeadm init --config kubeadm.yaml
```

/cc @luxas @mikedanese @pires

**Comment by luxas** [comment]
@mikedanese this was exactly what I was trying to say in https://github.com/kubernetes/kubernetes/pull/43835#issuecomment-290531847, however was mobile when travelling so hadn't all available context.

Do we really want to break self-hosting in v1.6.1?

I'm favor of marking v1.6.1-beta.0-5+abcdef with your patch stable in the deb/rpm

**Comment by mikedanese** [comment]
This is fixed.

**Comment by hwinkel** [comment]
which version of kubeadm should work then?

**Comment by luxas** [comment]
@hwinkel probably master.
Please open a new issue about this and the v1.6.1 version of kubeadm, we could see if we could fix self-hosting on the v1.6.1 branch by cherrypicking or so -- cc @andrewrynhard
